actionpack/test/dispatch/debug_exceptions_test.rb

```diff
@@ -620,4 +620,23 @@ def call(env)
       assert_select 'input[value="Action 2"]'
     end
   end
+
+  test "debug exceptions app shows diagnostics when malformed query parameters are provided" do
+    @app = DevelopmentApp
+
+    get "/bad_request?x[y]=1&x[y][][w]=2"
+
+    assert_response 400
+    assert_match "ActionController::BadRequest", body
+  end
+
+  test "debug exceptions app shows diagnostics when malformed query parameters are provided by xhr" do
```

Comment:
As per https://guides.rubyonrails.org/api_documentation_guidelines.html#wording . I think it should be written as `xhr` -> `XHR`.
 
```suggestion
  test "debug exceptions app shows diagnostics when malformed query parameters are provided by XHR" do
```